runtime/dict.go

```diff
@@ -407,22 +407,9 @@ func (d *Dict) putItem(f *Frame, key, value *Object) (*Object, *BaseException) {
 			raised = f.RaiseType(RuntimeErrorType, "dictionary changed during write")
 		} else {
 			if value == nil {
-				// Going to delete the entry.
-				if entry != nil && entry != deletedEntry {
-					d.table.storeEntry(index, deletedEntry)
-					d.table.incUsed(-1)
-					d.incVersion()
-				}
-			} else {
-				newEntry := &dictEntry{hash.Value(), key, value}
-				if newTable, ok := t.writeEntry(f, index, newEntry); ok {
-					if newTable != nil {
-						d.storeTable(newTable)
-					}
-					d.incVersion()
-				} else {
-					raised = f.RaiseType(OverflowErrorType, errResultTooLarge)
-				}
+				deleteDictEntry(d, entry, index)
```

Comment:
Would you mind also moving the code back into putItem()? Generally I prefer to having code inline rather than adding helpers that are only called in one place.